crates/swc_ecma_transforms_compat/tests/fixture/async-to-generator/issue-2556/1/exec.js

```diff
@@ -0,0 +1,17 @@
+async function init() {
+    return fib;
+    async function fib(n) {
+        if (n <= 1) {
+            return n;
+        }
+        const x = await fib(n - 1);
```

Comment:
Hi, @kdy1 ,
I added two tests with recursive calls.
Is that what you mean by `a function that calls itself` ?

Reply:
Yes, but I missed one.
Can you add a variable named `fib`?
Renaming bug will be triggered by `hygiene` and `hygiene` requires a conflicting variable.

Reply:
> Yes, but I missed one.
> Can you add a variable named `fib`?
> Renaming bug will be triggered by `hygiene` and `hygiene` requires a conflicting variable.

Where should I add it?
Is it below the `return fib` statement?

Reply:
I meant a global variable.

Reply:
It seems that the translated code works well. No naming conflicts.